### PR TITLE
Less long

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Profiler state that two costly part of getCard are querying the database and long -> Long. Actually, long->Long is not cached. So I change variables from long to Long. On my collection, it reduces the time from 23 seconds to 8 seconds in Long.valueOf

Furthermore, I change the query by moving in it a condition that it can execute. This reduce the time spent doing getLong from 6 to 5 seconds(I'll admit I was expecting a better improvement here)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java [hunk 1]
```diff
@@ -797,7 +797,7 @@ public void _remNotes(java.util.Collection<Long> ids) {
                     Timber.v("Empty card cancelled");
                     return null;
                 }
-                long nid = cur.getLong(0);
+                Long nid = cur.getLong(0);
                 long mid = cur.getLong(1);
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:801]: Was this one necessary, or missed?

**Review by david-allison-1 — APPROVED** [review]
Looks good - one query

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java (hunk 1 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:801]: `mid` is used only once, and it is used as a `long`. So in this case, we need `long`

**Review by mikehardy — APPROVED** [review]
Nice! I appreciate the profiler stats in the PRs, thanks @Arthur-Milchior

**Comment by Arthur-Milchior** [comment]
Note those are stats with the profiler on. Profiler makes everything slower. So the gain is not actually so big.

**Pull request merged**